### PR TITLE
fix xss vulnerability

### PR DESCRIPTION
**Pull request by alesevero**
This PR includes two middlewares for signatures validation:
- HTML clean up against XSS attacks through `HTMLPurifier`
  - Didn't need to use regex as `HTMLPurifier` wasn't cleaning up everything because we needed to decode the signature before; 
- HTML syntax check validation is not needed since `HTMLPurifier` cleans invalid HTML.

### CODE REVIEW

File: routes/routes.php [hunk 1]
```diff
@@ -86,12 +86,20 @@ function () {
...
                     \Railroad\Railforums\Controllers\UserForumSignaturesController::class.'@store'
                 )
+                    ->middleware([
+                        \Railroad\Railforums\Middleware\HTMLSanitization::class,
+                        \Railroad\Railforums\Middleware\SignatureHTMLSyntaxCheck::class
+                    ])
                     ->name('railforums.signature.store');
 
                 Route::patch(
                     'signature/update/{id}',
```
Comment (robwillems) [inline routes/routes.php:95]: There's 3 similar routes for update signature, are these used?  Does the app also allow you do update the signature?
image

File: src/Middleware/SignatureHTMLSyntaxCheck.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Railroad\Railforums\Middleware;
+
+use DOMDocument;
+use Illuminate\Http\Request;
+use Closure;
+
+class SignatureHTMLSyntaxCheck
```
Comment (robwillems) [inline src/Middleware/SignatureHTMLSyntaxCheck.php]: Is this check even necessary?  I can't find anything that gets through the sanitization check that causes this to fail.

**Review by robwillems — APPROVED** [review]
I think it looks great.

The purifier does change the syntax slightly so it may create some weird cases, but i think we'll just have to go with it and see.  You could script something to verify existing signatures, but it may not be worth it.

I would push this to a web-staging environment or beta-testing and pass some info along to Jon C for him to try it out and see what he thinks.

File: routes/routes.php (hunk 1 above)
Comment (alesevero) [reply routes/routes.php:95]: I don't think the app allows us to create/edit the signature. However, I've updated the API routes to include the middleware for consistency

File: src/Middleware/SignatureHTMLSyntaxCheck.php (hunk 2 above)
Comment (alesevero) [reply src/Middleware/SignatureHTMLSyntaxCheck.php]: Removed as the sanitization process removes invalid HTML as well

**Pull request merged**